src/main/kotlin/com/example/ticketbookingapp/webApi/screening/WebApiMovieScreeningController.kt

```diff
@@ -0,0 +1,46 @@
+﻿package com.example.ticketbookingapp.webApi.screening
+
+import com.example.ticketbookingapp.repositories.MovieScreeningRepository
+import com.example.ticketbookingapp.service.MovieScreeningSeatsService
+import com.example.ticketbookingapp.utils.findByIdOrClientError
+import com.example.ticketbookingapp.webApi.dto.MovieScreeningDto
+import com.example.ticketbookingapp.webApi.screening.dto.MovieScreeningAndRoomDto
+import com.example.ticketbookingapp.webApi.screening.dto.ScreeningInfoRequestDto
+import com.example.ticketbookingapp.webApi.screening.dto.SeatDto
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping(
+    path = ["/api"],
+    produces = ["application/json"]
+)
+// cors would be changed in final app
+@CrossOrigin(origins = ["*"])
+class WebApiMovieScreeningController(
+    private val seatsService: MovieScreeningSeatsService,
+    private val screeningRepository: MovieScreeningRepository,
+) {
+
+    @GetMapping("/screening")
+    @Transactional(rollbackFor = [Exception::class], readOnly = true)
+    fun getScreeningInfo(dto: ScreeningInfoRequestDto): MovieScreeningAndRoomDto {
```

Comment:
Trochę już późno :) Ale czy ja dobrze rozumiem, że w get przesyłasz payload z informacją o id screeningu?

Reply:
Tak. Nie jestem pewien czy o to tutaj chodzi, natomiast można to by uprościć pisząc argument
@RequestParam screeningId: Long
i wyrzucając klasę ScreeningInfoRequestDto